### PR TITLE
Reduce pylint warnings in isis_reduction_steps.py

### LINKED ISSUE #14738: Crunch pylint violations further for isis_reduction_steps.py
The top pylint offender file `isis_reduction_steps.py` will been reduced to 74 from around 145 once #14720 is merged , unfortunately it is still contains the most number of pylint violations,  these violations can further be crunched by wiping out two of the warning within from this file; `(unused-argument)` & `(redefined-buildin)`. These are fairly simple to fix, however there are many system tests which seem to be failing as soon as these fixes are made. Should be straight forward issue for someone who is familiar with the `isis_reduction_steps.py` file. This [wiki page](http://www.mantidproject.org/How_to_run_Pylint#PyCharm_-_JetBrains) can be used to set up pylint locally, in order to quickly view the violations within PyCharm.

### LINKED ISSUE #15107: Reduce Pylint warnings for isis_reduction_steps.py
***This is a sub-issue of*** #14705 

Reduce the number of Pylint warnings in the isis_reduction_steps.py file

### PR DESCRIPTION
**Pull request by MattKing06**
**To Test:**
- Code review
- All tests should pass
- See that PyLint build reports less warnings for isis_reduction_steps.py than previously

**Release Notes**
This is an internal maintenance issue with no impact on Mantid itself, so no update is needed for release notes.

Fixes #15107 
Fixes #14738

### CODE REVIEW

File: scripts/SANS/isis_reduction_steps.py [hunk 1]
```diff
@@ -1017,9 +1025,9 @@ def execute(self, reducer, workspace):
         # reset the xml, as execute can be run more than once
         self._xml = []
 
-        if (not self.min_radius is None) and (self.min_radius > 0.0):
+        if (self.min_radius is not None) and (self.min_radius > 0.0):
             self.add_cylinder(self.min_radius, 0, 0, 'beam_stop')
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: Good

File: scripts/SANS/isis_reduction_steps.py [hunk 2]
```diff
@@ -1066,13 +1074,13 @@ def view(self, instrum):
         instrum.load_empty(wksp_name)
 
         # apply masking to the current detector
-        self.execute(None, wksp_name, instrum)
+        self.execute(None, wksp_name)
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: Do you know how this worked?

File: scripts/SANS/isis_reduction_steps.py [hunk 3]
```diff
@@ -1758,11 +1766,12 @@ def _evaluate_settings(self):
...
-        if (not has_max_one_run_number(indices_time_detector) or
-            not has_max_one_run_number(indices_time_monitor) or
-            not has_max_one_run_number(indices_uamp_detector) or
-            not has_max_one_run_number(indices_uamp_monitor) ) :
-            raise RuntimeError("DarkRunSubtraction: More background correction runs have been specified than are allowed. "
+        if not has_max_one_run_number(indices_time_detector) or \
+           not has_max_one_run_number(indices_time_monitor) or \
+           not has_max_one_run_number(indices_uamp_detector) or \
+           not has_max_one_run_number(indices_uamp_monitor):
+            raise RuntimeError("DarkRunSubtraction: More background correction runs have been "
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: Was that a pylint warning? Braces are a valid way of having multi-line expressions in python. But if pylint says that the `\` approach is preferred, then that is fine.

File: scripts/SANS/isis_reduction_steps.py [hunk 4]
```diff
@@ -1797,7 +1806,7 @@ def _get_final_setting_detectors(self, run_number, use_mean, use_time, indices):
         detector_mean = [use_mean[index] for index in indices]
         detector_time = [use_time[index] for index in indices]
 
-        if len(detector_runs) == 0 or detector_runs[0] == None:
+        if len(detector_runs) == 0 or detector_runs[0] is None:
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: Good

File: scripts/SANS/isis_reduction_steps.py [hunk 5]
```diff
@@ -2115,40 +2125,44 @@ def setup_wksp(self, inputWS, inst, wavbining, trans_det_ids, reducer):
...
-                WorkspaceIndexList=ws_index,
-                Mode='Mean',
-                SkipMonitors=True)
-
-        ConvertUnits(InputWorkspace=tmpWS, OutputWorkspace=tmpWS, Target="Wavelength")
-
-        if self.interpolate:
-            InterpolatingRebin(InputWorkspace=tmpWS, OutputWorkspace=tmpWS, Params=wavbining)
+        if tmp.getNumberHistograms() > 0:
+            ws_index = 0
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: Why did you add this?
1. It is good to test against empty workspaces.
2. I don't think that the `ws_index` variable needs to be initialized like this.

Let's discuss this in person maybe.

File: scripts/SANS/isis_reduction_steps.py [hunk 6]
```diff
@@ -2442,7 +2457,9 @@ class CalculateNormISIS(object):
     WAVE_CORR_NAME = '__Q_WAVE_conversion_temp'
     PIXEL_CORR_NAME = '__Q_pixel_conversion_temp'
 
-    def __init__(self, wavelength_deps=[]):
+    def __init__(self, wavelength_deps=None):
+        if wavelength_deps is None:
+            wavelength_deps = []
```
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: These measures are very good.

File: scripts/SANS/isis_reduction_steps.py (hunk 5 above)
Comment (MattKing06) [inline scripts/SANS/isis_reduction_steps.py]: Pylint was complaining that ws_index could not be initialized (in the case that getNumberHistograms() returned 0). For some reason, if I only included the if statement and did not set ws_index to 0 then Pylint was still complaining.

File: scripts/SANS/isis_reduction_steps.py (hunk 2 above)
Comment (MattKing06) [inline scripts/SANS/isis_reduction_steps.py]: the execute method for this class takes 2 arguments (`reducer` and  `workspace`). Instead of passing through the instrument name, I thought it was more appropriate to pass through the workspace name to the execute method. The instrument is also retrieved the reducer argument in the execute method any way.

File: scripts/SANS/isis_reduction_steps.py (hunk 5 above)
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: I see, the ws_index is used below, hence it requires initialization. Although I seem to have touched that area, I don't quite understand yet why this is required (nothing to do with this PR though).

File: scripts/SANS/isis_reduction_steps.py (hunk 2 above)
Comment (picatess) [inline scripts/SANS/isis_reduction_steps.py]: hmm, that does not explain how we get away with passing in three parameters.  I checked and the base object has an execute with three [parameters](https://github.com/mantidproject/mantid/blob/master/scripts/reducer_singleton.py#L31:L38). It would default to it, but it should then throw an exception. Might need more investigation.

**Comment by MattKing06** [comment]
@picatess can this be reviewed by a gatekeeper now? or are there more improvements to be made?

**Comment by picatess** [comment]
:shipit:

**Comment by FedeMPouzols** [comment]
Good to get rid of quite a few more pylint issues. I'll update the pylint threshold in Jenkins when I see the new number of warnings.
